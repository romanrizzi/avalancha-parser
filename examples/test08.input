
-- Conjunci√≥n

check A and B 
check A and B and C
check A and B and C and D
check not A and not B and not C and not D

check A1==A2 and B1==B2
check A1==A2 and B1==B2 and C1==C2
check A1==A2 and B1==B2 and C1==C2 and D1==D2
check not A1==A2 and not B1==B2 and not C1==C2 and not D1==D2

check A and B 
check (A and B) and C
check A and (B and C) and D
check not ((A and not (B and not C)) and not D)

check A1==A2 and B1==B2
check A1==A2 and (B1==B2 and C1==C2)
check ((A1==A2 and B1==B2) and C1==C2) and D1==D2
check not (A1==A2 and not (B1==B2 and not (C1==C2 and not (D1==D2))))

