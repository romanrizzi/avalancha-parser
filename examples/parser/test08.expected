[
   "program",
   [],
   [
      [
         "check",
         [
            "and",
            [
               "equal",
               [
                  "cons",
                  "A",
                  []
               ],
               [
                  "cons",
                  "True",
                  []
               ]
            ],
            [
               "equal",
               [
                  "cons",
                  "B",
                  []
               ],
               [
                  "cons",
                  "True",
                  []
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "and",
            [
               "equal",
               [
                  "cons",
                  "A",
                  []
               ],
               [
                  "cons",
                  "True",
                  []
               ]
            ],
            [
               "and",
               [
                  "equal",
                  [
                     "cons",
                     "B",
                     []
                  ],
                  [
                     "cons",
                     "True",
                     []
                  ]
               ],
               [
                  "equal",
                  [
                     "cons",
                     "C",
                     []
                  ],
                  [
                     "cons",
                     "True",
                     []
                  ]
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "and",
            [
               "equal",
               [
                  "cons",
                  "A",
                  []
               ],
               [
                  "cons",
                  "True",
                  []
               ]
            ],
            [
               "and",
               [
                  "equal",
                  [
                     "cons",
                     "B",
                     []
                  ],
                  [
                     "cons",
                     "True",
                     []
                  ]
               ],
               [
                  "and",
                  [
                     "equal",
                     [
                        "cons",
                        "C",
                        []
                     ],
                     [
                        "cons",
                        "True",
                        []
                     ]
                  ],
                  [
                     "equal",
                     [
                        "cons",
                        "D",
                        []
                     ],
                     [
                        "cons",
                        "True",
                        []
                     ]
                  ]
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "and",
            [
               "not",
               [
                  "equal",
                  [
                     "cons",
                     "A",
                     []
                  ],
                  [
                     "cons",
                     "True",
                     []
                  ]
               ]
            ],
            [
               "and",
               [
                  "not",
                  [
                     "equal",
                     [
                        "cons",
                        "B",
                        []
                     ],
                     [
                        "cons",
                        "True",
                        []
                     ]
                  ]
               ],
               [
                  "and",
                  [
                     "not",
                     [
                        "equal",
                        [
                           "cons",
                           "C",
                           []
                        ],
                        [
                           "cons",
                           "True",
                           []
                        ]
                     ]
                  ],
                  [
                     "not",
                     [
                        "equal",
                        [
                           "cons",
                           "D",
                           []
                        ],
                        [
                           "cons",
                           "True",
                           []
                        ]
                     ]
                  ]
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "and",
            [
               "equal",
               [
                  "cons",
                  "A1",
                  []
               ],
               [
                  "cons",
                  "A2",
                  []
               ]
            ],
            [
               "equal",
               [
                  "cons",
                  "B1",
                  []
               ],
               [
                  "cons",
                  "B2",
                  []
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "and",
            [
               "equal",
               [
                  "cons",
                  "A1",
                  []
               ],
               [
                  "cons",
                  "A2",
                  []
               ]
            ],
            [
               "and",
               [
                  "equal",
                  [
                     "cons",
                     "B1",
                     []
                  ],
                  [
                     "cons",
                     "B2",
                     []
                  ]
               ],
               [
                  "equal",
                  [
                     "cons",
                     "C1",
                     []
                  ],
                  [
                     "cons",
                     "C2",
                     []
                  ]
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "and",
            [
               "equal",
               [
                  "cons",
                  "A1",
                  []
               ],
               [
                  "cons",
                  "A2",
                  []
               ]
            ],
            [
               "and",
               [
                  "equal",
                  [
                     "cons",
                     "B1",
                     []
                  ],
                  [
                     "cons",
                     "B2",
                     []
                  ]
               ],
               [
                  "and",
                  [
                     "equal",
                     [
                        "cons",
                        "C1",
                        []
                     ],
                     [
                        "cons",
                        "C2",
                        []
                     ]
                  ],
                  [
                     "equal",
                     [
                        "cons",
                        "D1",
                        []
                     ],
                     [
                        "cons",
                        "D2",
                        []
                     ]
                  ]
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "and",
            [
               "not",
               [
                  "equal",
                  [
                     "cons",
                     "A1",
                     []
                  ],
                  [
                     "cons",
                     "A2",
                     []
                  ]
               ]
            ],
            [
               "and",
               [
                  "not",
                  [
                     "equal",
                     [
                        "cons",
                        "B1",
                        []
                     ],
                     [
                        "cons",
                        "B2",
                        []
                     ]
                  ]
               ],
               [
                  "and",
                  [
                     "not",
                     [
                        "equal",
                        [
                           "cons",
                           "C1",
                           []
                        ],
                        [
                           "cons",
                           "C2",
                           []
                        ]
                     ]
                  ],
                  [
                     "not",
                     [
                        "equal",
                        [
                           "cons",
                           "D1",
                           []
                        ],
                        [
                           "cons",
                           "D2",
                           []
                        ]
                     ]
                  ]
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "and",
            [
               "equal",
               [
                  "cons",
                  "A",
                  []
               ],
               [
                  "cons",
                  "True",
                  []
               ]
            ],
            [
               "equal",
               [
                  "cons",
                  "B",
                  []
               ],
               [
                  "cons",
                  "True",
                  []
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "and",
            [
               "and",
               [
                  "equal",
                  [
                     "cons",
                     "A",
                     []
                  ],
                  [
                     "cons",
                     "True",
                     []
                  ]
               ],
               [
                  "equal",
                  [
                     "cons",
                     "B",
                     []
                  ],
                  [
                     "cons",
                     "True",
                     []
                  ]
               ]
            ],
            [
               "equal",
               [
                  "cons",
                  "C",
                  []
               ],
               [
                  "cons",
                  "True",
                  []
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "and",
            [
               "equal",
               [
                  "cons",
                  "A",
                  []
               ],
               [
                  "cons",
                  "True",
                  []
               ]
            ],
            [
               "and",
               [
                  "and",
                  [
                     "equal",
                     [
                        "cons",
                        "B",
                        []
                     ],
                     [
                        "cons",
                        "True",
                        []
                     ]
                  ],
                  [
                     "equal",
                     [
                        "cons",
                        "C",
                        []
                     ],
                     [
                        "cons",
                        "True",
                        []
                     ]
                  ]
               ],
               [
                  "equal",
                  [
                     "cons",
                     "D",
                     []
                  ],
                  [
                     "cons",
                     "True",
                     []
                  ]
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "not",
            [
               "and",
               [
                  "and",
                  [
                     "equal",
                     [
                        "cons",
                        "A",
                        []
                     ],
                     [
                        "cons",
                        "True",
                        []
                     ]
                  ],
                  [
                     "not",
                     [
                        "and",
                        [
                           "equal",
                           [
                              "cons",
                              "B",
                              []
                           ],
                           [
                              "cons",
                              "True",
                              []
                           ]
                        ],
                        [
                           "not",
                           [
                              "equal",
                              [
                                 "cons",
                                 "C",
                                 []
                              ],
                              [
                                 "cons",
                                 "True",
                                 []
                              ]
                           ]
                        ]
                     ]
                  ]
               ],
               [
                  "not",
                  [
                     "equal",
                     [
                        "cons",
                        "D",
                        []
                     ],
                     [
                        "cons",
                        "True",
                        []
                     ]
                  ]
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "and",
            [
               "equal",
               [
                  "cons",
                  "A1",
                  []
               ],
               [
                  "cons",
                  "A2",
                  []
               ]
            ],
            [
               "equal",
               [
                  "cons",
                  "B1",
                  []
               ],
               [
                  "cons",
                  "B2",
                  []
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "and",
            [
               "equal",
               [
                  "cons",
                  "A1",
                  []
               ],
               [
                  "cons",
                  "A2",
                  []
               ]
            ],
            [
               "and",
               [
                  "equal",
                  [
                     "cons",
                     "B1",
                     []
                  ],
                  [
                     "cons",
                     "B2",
                     []
                  ]
               ],
               [
                  "equal",
                  [
                     "cons",
                     "C1",
                     []
                  ],
                  [
                     "cons",
                     "C2",
                     []
                  ]
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "and",
            [
               "and",
               [
                  "and",
                  [
                     "equal",
                     [
                        "cons",
                        "A1",
                        []
                     ],
                     [
                        "cons",
                        "A2",
                        []
                     ]
                  ],
                  [
                     "equal",
                     [
                        "cons",
                        "B1",
                        []
                     ],
                     [
                        "cons",
                        "B2",
                        []
                     ]
                  ]
               ],
               [
                  "equal",
                  [
                     "cons",
                     "C1",
                     []
                  ],
                  [
                     "cons",
                     "C2",
                     []
                  ]
               ]
            ],
            [
               "equal",
               [
                  "cons",
                  "D1",
                  []
               ],
               [
                  "cons",
                  "D2",
                  []
               ]
            ]
         ]
      ],
      [
         "check",
         [
            "not",
            [
               "and",
               [
                  "equal",
                  [
                     "cons",
                     "A1",
                     []
                  ],
                  [
                     "cons",
                     "A2",
                     []
                  ]
               ],
               [
                  "not",
                  [
                     "and",
                     [
                        "equal",
                        [
                           "cons",
                           "B1",
                           []
                        ],
                        [
                           "cons",
                           "B2",
                           []
                        ]
                     ],
                     [
                        "not",
                        [
                           "and",
                           [
                              "equal",
                              [
                                 "cons",
                                 "C1",
                                 []
                              ],
                              [
                                 "cons",
                                 "C2",
                                 []
                              ]
                           ],
                           [
                              "not",
                              [
                                 "equal",
                                 [
                                    "cons",
                                    "D1",
                                    []
                                 ],
                                 [
                                    "cons",
                                    "D2",
                                    []
                                 ]
                              ]
                           ]
                        ]
                     ]
                  ]
               ]
            ]
         ]
      ]
   ]
]
